docs-experimental/skopeo-experimental-image-proxy.1.md

```diff
@@ -0,0 +1,133 @@
+    *   Args: `[uint64 imageID, string digest]`
+    *   Returns: `int64` (blob size in `value`), blob data via main pipe, error status via error pipe.
+*   **GetLayerInfo**: Retrieves an array of objects, each describing an image layer (digest, size, mediaType). **Note**: This method returns data inline and may fail for images with many layers due to message size limits. Use `GetLayerInfoPiped` for a more robust solution.
+    *   Args: `[uint64 imageID]`
+    *   Returns: `array` of `{digest: string, size: int64, mediaType: string}`.
+*   **GetLayerInfoPiped**: Retrieves information about image layers, similar to `GetLayerInfo`. Layer information data is streamed over a pipe. (Introduced in protocol version 0.2.7)
+    *   Args: `[uint64 imageID]`
+    *   Returns: `null`, layer information data via pipe.
+*   **FinishPipe**: Signals that the client has finished reading all data from a pipe associated with a `pipeid` (obtained from methods like `GetManifest` or `GetBlob`). This allows the server to close its end of the pipe and report any pending errors (e.g., digest verification failure for `GetBlob`). This method **must** be called by the client after consuming data from a pipe, except for pipes from `GetRawBlob`.
+    *   Args: `[uint32 pipeID]`
+    *   Returns: `null`
+*   **Shutdown**: Instructs the proxy server to terminate gracefully.
+    *   Args: `[]` (empty array)
+    *   Returns: `null`
+
+**Data Transfer for Pipes**
+
+When a method returns a `pipeid`, the server also passes the read-end of a pipe via file descriptor (FD) passing. The client reads the data (e.g., manifest content, blob content) from this FD. After successfully reading all data, the client **must** call `FinishPipe` with the corresponding `pipeid`. This signals to the server that the transfer is complete, allows the server to clean up resources, and enables the client to check for any errors that might have occurred during the data streaming process (e.g., a digest mismatch during `GetBlob`). The `GetRawBlob` method is an exception; it uses a dedicated error pipe instead of the `FinishPipe` mechanism.
+
+# OPTIONS
+**--sockfd**=*fd*
+Serve on the opened socket passed as file descriptor *fd*. Defaults to 0 (standard input).
+
+The command also supports common skopeo options for interacting with image registries and local storage. These include:
+
+**--authfile**=*path*
+Path of the authentication file.
+Default is `${XDG_RUNTIME_DIR}/containers/auth.json`. If `XDG_RUNTIME_DIR` is not set, the default is `$HOME/.config/containers/auth.json`. If this file does not exist, `$HOME/.docker/config.json` is used.
```

Comment:
`RUNTIME_DIR` does _not_ default to `.config/…`. I guess just drop that sentence.

(In general it would be more convenient to maintain if the option descriptions were copy&pasted from peer man pages, and not idiosyncratic … I _don’t_ mean that the current wording is perfect and can’t be improved.)

Reply:
Yeah, I switched to copy-pasting more.